Automattic/vip-go-mu-plugins
# Don't instantiate the Jobs objects and don't try to schedule them in web context

## Description

Currently, we're running the job classes instantiation both on `admin_init` and in CLI .

There's no need to do so because Cron Control is regularly spawning cron processes, so there's not a scenario where running on `admin_init` is beneficial/needed.

## Changelog Description

### Plugin Updated: Enterprise Search

We've implemented minor improvements that will help minimize potential race conditions on the job scheduling.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test


1. Checkout the PR
2. Spin u the dev env `vip dev-env start`
3. Run `vip dev-env exec  -- wp cron event delete vip_search_settings_health`
4. Run `vip dev-env exec -- wp cron event list`
5. Verify event got rescheduled.
6. Run `wp shell`
7. Run ```var_dump( wp_clear_scheduled_hook( 'vip_search_settings_health' ) );```
Should get:
```
int(1)
```
8. Run step 7 again, should get `0` now.
9. Exit and spawn another wp command and verify it got scheduled
